code/_onclick/hud/alert.dm

```diff
@@ -870,3 +870,56 @@ so as to remain in compliance with the most up-to-date laws."
 	if(!istype(usr))
 		return
 	living_owner.do_succumb(TRUE)
+
+/atom/movable/screen/alert/changeling_defib_revive
+	name = "Your heart is being defibrillated."
+	desc = "Click this status to be revived from fake death."
+	icon_state = "template"
+	timeout = 10 SECONDS
+	var/do_we_revive = FALSE
+	var/image/toggle_overlay
+
+/atom/movable/screen/alert/changeling_defib_revive/Initialize(mapload, parent_unit,)
+	. = ..()
+	var/image/defib_appearance = image(parent_unit)
+	defib_appearance.layer = FLOAT_LAYER
+	defib_appearance.plane = FLOAT_PLANE
+	overlays += defib_appearance
+
+/atom/movable/screen/alert/changeling_defib_revive/attach_owner(mob/new_owner)
+	. = ..()
+	RegisterSignal(owner, COMSIG_LIVING_DEFIBBED, PROC_REF(on_defib_revive))
+
+/atom/movable/screen/alert/changeling_defib_revive/Destroy()
+	if(toggle_overlay)
+		overlays -= toggle_overlay
+		qdel(toggle_overlay)
+	return ..()
+
+/atom/movable/screen/alert/changeling_defib_revive/Click()
+	if(!..())
+		return
+	do_we_revive = !do_we_revive
+	if(!toggle_overlay)
+		toggle_overlay = image('icons/mob/screen_gen.dmi', icon_state = "selector")
+		toggle_overlay.layer = FLOAT_LAYER
+		toggle_overlay.plane = FLOAT_PLANE + 2
+	if(do_we_revive)
+		overlays |= toggle_overlay
+	else
+		overlays -= toggle_overlay
+
+/atom/movable/screen/alert/changeling_defib_revive/proc/on_defib_revive()
+	. = COMPONENT_DEFIB_FAKEDEATH_DENIED
+	if(do_we_revive)
+		var/datum/antagonist/changeling/cling = owner.mind.has_antag_datum(/datum/antagonist/changeling)
+		cling.remove_specific_power(/datum/action/changeling/revive)
+		cling.regenerating = FALSE
+		var/heart_name = "heart"
+		if(iscarbon(owner))
```

Comment:
Is there any chance that the owner would not be a carbon here?

Reply:
I don't think so, but I left it like that to be safe